### DISCUSSION TITLE
Synology NAS has changed from Docker to Container Manager.. Docker legacy CLI commands NO longer work

### DISCUSSION THREAD

**Discussion by Daekwan17** [teslamate-org/teslamate#3429] [General]
So looks like I've not touched my Teslamate instance which is hosted on my Synology NAS in a while.   Everything has been working fine for the last couple of years, well until last weekend.  For some reason, my car no longer provides any updates in Teslamate.   I tried to troubleshoot what was going on and started with the basic Docker backup of my Teslamate database only to find that none of the Docker commands worked anymore.   Apparently, Synology has changed from "Docker" to "Container Manager" with the latest DSM 7.2 software update, and Im having a hard time figuring anything out container-wise going forward.   The instructions to install and maintain the Docker containers no longer work with the new Container Manager upgrade.

Ideally, I'd to do three things:
1) Back up my existing Teslamate data.   I need help with what commands to use nothing docker-compose works anymore.
2) Delete all my existing Containers in Container Manager, so I can start fresh.
3) Install a new instance of Teslamate on my Synology NAS (or Windows PC).
4) Restore my existing Teslamate data.  And use the fresh installation going forward.

Any help would be greatly appreciated.  Thanks in advance!!

**Comment by rndpsn** [comment]
I'm on DSM 7.2-64570 and everything still works for me.  I'm using the docker-compose command to back up:

`root@dsm:/volume1/@docker# docker-compose exec -T database pg_dump -U teslamate teslamate > /volume1/backupfile.bck`

I also have a nightly job on DSM that does a docker-compose pull and docker-compose up to update Teslamate, and the job hasn't changed with DSM 7.2.x.

**Reply by Techtobi83** [reply]
Thats interesting, are you able to explain what this command does?  I am not a console-happy-user, i think you create a kind of compressed file including complete content of a folder? 
How does this work automatically?

With the exception of Teslamate, I work only via GUI on Synology and all my installations have been installed in folder structures that are visible to me. Unfortunately, except for Teslamate, for which I don't have an isolated backup strategy (except that the entire Synology is mirrored once a week)

**Reply by rndpsn** [reply]
The backup command came from the docs. I created a Scheduled Task in DSM to run it daily with an edited version of the command to add the date stamp to the file name.

**Reply by rndpsn** [reply]
More details:

DSM Control Panel > Task Scheduler, Create > Scheduled Task > User-defined script, for user "root":

This is my script. 

Line 1 is the actual backup, putting the resulting file in the directory "/volume1/backups/TeslaMate/" with a date stamp in the file name eg. "TeslaMateDB_20231115.bck".
Line 2 deletes older backup files s older than 3 days (today minus 2 days).
Line 3 list all the files in the backup directory (I've "Send run details by email" enabled for the task).

```
/usr/local/bin/docker-compose --file /volume1/@docker/docker-compose.yml exec -T database pg_dump -U teslamate teslamate > /volume1/backups/TeslaMate/TeslaMateDB_$(/bin/date +%Y%m%d).bck;

/bin/find /volume1/backups/TeslaMate -maxdepth 1 -type f -name '*.bck' -mtime +2 -exec /bin/rm -f '{}' \;

/bin/ls -alth /volume1/backups/TeslaMate/;
```

**Comment by Daekwan17** [comment]
Thank you so much for posting that command.  I copied your command and it worked immediately!!

Appears the instructions on the docs.telsamate.org are missing the dash in between the words docker and compose.  So instead of "docker-compose", it is written as "docker compose".   And I was just copy & pasting the command lol.   
https://docs.teslamate.org/docs/maintenance/backup_restore

Of course now I feel like such an idiot lol.

**Reply by rc-16** [reply]
I'm not an expert but I was reading https://docs.docker.com/compose/reference/ . It seems that the original docker-compose has been replaced by a version 2 that uses "docker compose" instead of "docker-compose". Have a read & see if you come to the same conclusion as me. Maybe the documentation refers to the newer version?

**Reply by reuten75** [reply]
Hi all experts :) I'm also trying to use the reference to make a backup but it wont work for me either. I am using this command:
docker compose exec -T database pg_dump -U teslamate teslamate > /volume1/docker/backup/teslamate.bck

And this is what happens:
unknown shorthand flag: 'T' in -T

It creates the file but it is empty.
What is wrong? Tried both database and db.

**Reply by ftt1211** [reply]
> Hi all experts :) I'm also trying to use the reference to make a backup but it wont work for me either. I am using this command: docker compose exec -T database pg_dump -U teslamate teslamate > /volume1/docker/backup/teslamate.bck
> 
> And this is what happens: unknown shorthand flag: 'T' in -T
> 
> It creates the file but it is empty. What is wrong? Tried both database and db.

I get the same unkonwn shorthand flag message. Did you figure out what was wrong?

**Comment by Daekwan17** [comment]
Well, the backup command did successfully complete.  However, Im back to my original issue where my car is still not updating its status with Teslamate.   Its either been showing as "unavailable" or "asleep" for the past 4 days.   If I go to the "States" dashboard in Grafana, the only thing I see for status is the green sleep status going all the way back to 11/10/23 at 18:32:18 time mark.  Which marks the last time the car showed up as "online" with a blue status.

This is the current Teslamate dashboard status:

Model Y 50

Status	asleep for 1 d, 21 h
Range (rated)	92.7 mi
Range (est.)	102.2 mi
State of Charge	41%
Mileage	52976 mi
Version	[2023.38.6](https://www.notateslaapp.com/software-updates/version/2023.38.6/release-notes)

Of course, I've been driving and charging the car every day since that time mark.   I did resync my API and Refresh tokens with Teslamate using the Tesla Auth application.   It appears something is blocking the data transfer from the Tesla's servers to the vehicles.  I wonder if anything has changed with 3rd party API access to Tesla's data.

**Reply by rndpsn** [reply]
Hi Daekwan, 

That should be another ticket in order to get the proper attention; sorry can't help with that issue.

**Reply by cwanja** [reply]
What version of TeslaMate are you running? Based upon your original comment, it seems you might not have updated TeslaMate in a while.

**Comment by Daekwan17** [comment]
First of all.. thank you everyone!!   After I was able to successfully backup the Teslamate database.. I just went ahead and deleted the entire Container Manager package and deleted all the directories related to it.  I also deleted a bunch of other Synology packages that were problematic or just not being used.

Looks like the issue was actually my Synology NAS ran out of ALL available space on the system partition (/dev/md0).   I believe the Synology package Active Insight was the culprit as it created 700MB worth of files on the system partition and nothing else was available to write there.   With the system partition now available to store data again.. I then reinstalled Container Manager and re-intalled Teslamate using the guide provided here:  https://docs.teslamate.org/docs/installation/docker

After the initial setup.. I made sure I could see my car actively reporting data.  Then I followed the instructions to restore the database from here:  https://docs.teslamate.org/docs/maintenance/backup_restore/   I am extremely happy to say that everything is back to normal and I can see all of my historical data from the past 2 years.    I did lose the data from the past 4 days, but Im happy I caught the issue before any other data collection was missed.

Many thanks to everyone.  And to answer the final question I am running Teslamate v1.27.4 and my car is running 2023.38.6.   
<body>

2023.38.6
--



</body>
</html>[2023.38.6](https://www.notateslaapp.com/software-updates/version/2023.38.6/release-notes)

**Comment by Daekwan17** [comment]
Yep. Use docker-compose instead of docker compose. Sent from my iPhoneOn Jul 7, 2024, at 5:20 PM, ftt1211 ***@***.***> wrote:﻿

Hi all experts :) I'm also trying to use the reference to make a backup but it wont work for me either. I am using this command: docker compose exec -T database pg_dump -U teslamate teslamate > /volume1/docker/backup/teslamate.bck
And this is what happens: unknown shorthand flag: 'T' in -T
It creates the file but it is empty. What is wrong? Tried both database and db.

I get the same unkonwn shorthand flag message. Did you figure out what was wrong?

**Reply by ftt1211** [reply]
Thanks. I think it was also the fact that although i was SSHed in as Admin, it needed to be root...which sudo -i command allowed me to go into root.
